protocol/avdl/keybase1/gui_file_access.avdl

```diff
@@ -0,0 +1,129 @@
+@namespace("keybase.1")
+
+protocol GuiFileAccess {
+
+  import idl "common.avdl";
+
+/*
+  File manipulation interface
+  
+  Caller provides OpID to asynchronous method call, repeats until status is no
+  longer pending or operation is cancelled.
+ 
+  KBFS paths are expected to start with "private" or "public" - mount name is implied.
+*/
+
+  // Random GUID provided by caller 
+  fixed OpID(16);
+  
+  enum ItemType {
+    FILE,
+    DIR
+  }
+
+  record Item {
+    string path;
+    Time time;
+    int size;   // size of file or count of dir entries
+    ItemType itemType;
+  }
+
+  enum Status {
+    finished,
+    pending,
+    invalidated,
+    error
+  }
+
+  record Result {
+    Status status;
+    int progress;       // if status == pending
+    int errno;          // if status == error
+    string description; // if status == error
+  }
+
+  record ListResult {
+    array<Item> items;
+    Result result;
+  }
+
+  record FileContent {
+    bytes data
+    Result result;
+  }
+
+  /**
+   Begin list if items in directory at path
+   Return indicates error
+   Retrieve results with getList()
+   */
+  int startList(OpID opID, string path);
+
+  /**
+   Begin recursive list if items in directory at path
+   Return indicates error
+   */
+  int startListRecursive(OpID opID, string path);
+
+  /**
+   Get list if items in progress. Can indicate status of pending
+   to get more items.
+   */
+  ListResult getList(OpID opID);
+
+  /**
+   Begin copy of file or directory into filesystem 
+   */
+  int startCopyIn(OpID opID, string src, string dest);
```

Comment:
Why is there a need for `copyIn` versus `copyOut`?  Why not just `copy()` and `copyRecursive`, and then have the service figure it out from there. Otherwise, it's going to explode into {in,out,inside}X{recursive,nonRecursive}, or 6 API calls.  Better just to have 2.